rtfs.go

```diff
@@ -155,3 +158,40 @@ func (im *IpfsManager) CustomRequest(ctx context.Context, url, commad string,
 func (im *IpfsManager) SwarmConnect(ctx context.Context, addrs ...string) error {
 	return im.shell.SwarmConnect(ctx, addrs...)
 }
+
+// DedupAndCalculatePinSize is used to remove duplicate refers to objects for a more accurate pin size cost
+// it returns the size of all refs, as well as all unique references
+func (im *IpfsManager) DedupAndCalculatePinSize(hash string) (int64, []string, error) {
```

Comment:
I wouldn’t have made this a class function - since it’s sort of an extended API, we don’t need to keep growin the IpfsManager interface to accommodate misc helpers

It would instead be a function that accepts IpfsManager as a parameter - we should minimize the interface as much as possible

Reply:
Ah yes that's a very good design choice. Once I get Temporal properly behaving with the massive update this change brings into dependencies I'll re-work this and open a new PR